### DISCUSSION TITLE
CH04_01 앤서블 소개 강의 중에 테라폼 remote_state의 organization과 workspace 설정 및 terraform apply 오류 문의

### DISCUSSION THREAD

**Discussion by Virusuki** [tedilabs/fastcampus-devops#55] [Q&A] [answered]
안녕하세요 강사님,
좋은 강의 감사드립니다.
CH04_01 앤서블 소개 강의 중에 테라폼 remote_state의 organization과 workspace를 설정하고 terraform apply를 진행 과정에서 다음과 같이 오류가 발생해서요.
![테라폼오류1](https://user-images.githubusercontent.com/70243256/155873486-8da312fc-0a7e-40ba-b55a-ef459f15d99c.PNG)

그리고, config.yaml의 Organization과 workspace 설정을 위해서,,, 테라폼 홈페이지에 Organization과 workspace를 셋팅과 환경변수 셋업을 강의보고 다시 진행했는데 이렇게 해서 config.yaml에 설정하는게 맞는지 문의드려요.
그리고, variables 환경변수에 AWS_ACCESS_KEY_ID의 값은
aws sts get-caller-identity명령어에서 UserId": "AIDASAVVASNLJQCW63VAE 값을 기입했는데,,, 확인 부탁드리겠습니다.!!

![테라폼 remote 셋업](https://user-images.githubusercontent.com/70243256/155873587-4d0055fe-6134-4d74-a368-2794ddad5a00.PNG)

------------------------------------------------------------------------------------------------------------------------------------------------

강사님,,, 다시 몇가지 수정했는데
테라폼 클라우드 에서는 그림과 같습니다.
![terraform_remote](https://user-images.githubusercontent.com/70243256/155905781-aaa0185d-3ef8-4e4f-b226-d2acccde7365.PNG)

그리고, 다시 terraform apply 하니

![unable_remote_state](https://user-images.githubusercontent.com/70243256/155905812-be4c76f4-7c43-45d5-9051-91a263f59f96.PNG)

Error : Unable to find remote state 에러인데,,, 원인을 못찾고 있습니다. ㅠㅠ

**Comment by posquit0** [comment, marked as answer]
안녕하세요! 제가 개인 사정으로 지방에 내려와 있어서 답변이 늦었습니다. :)

- `fastcampus-devops/2-ansible/lab/network` 디렉토리에서 우선 네트워크 테라폼 워크스페이스를 `terraform init`, `terraform apply`를 수행하시고 나서 진행하시면 됩니다.
- `remote_states`는 강의에서 이야기하는 것처럼 다른 테라폼 워크스페이스의 결과물을 데이터로 참조하기 위함입니다. 때문에 해당 워크스페이스가 미리 만들어져 있어야 합니다.
- 제 ansible / terraform / packer 에서 사용하는 테라폼 코드는 모두 동일한 네트워크 테라폼 워크스페이스를 사용하니 참고 부탁드립니다.
